tests/routes/test_geostore.py

```diff
@@ -0,0 +1,159 @@
+from typing import List
+from uuid import UUID
+
+import pytest
+
+from app.application import ContextEngine, db
+from app.models.orm.geostore import Geostore
+from app.models.pydantic.geostore import GeostoreResponse
+from tests import BUCKET, GEOJSON_NAME
+from tests.utils import create_default_asset
+
+
+@pytest.mark.asyncio
+async def test_user_area_geostore(async_client):
+    # This is the gfw_geostore_id returned when POSTing the payload with Postman
+    expected_goestore_id = "b44a9213-4fc2-14e6-02e3-96faf0d89499"
+
+    # Start off by checking the response for a geostore that doesn't exist
+    resp = await async_client.get(f"/geostore/{expected_goestore_id}")
+    assert resp.status_code == 404
+    assert resp.json() == {
+        "status": "failed",
+        "data": f"Area with gfw_geostore_id {expected_goestore_id} does not exist",
+    }
+
+    payload = {
+        "type": "FeatureCollection",
+        "features": [
+            {
+                "type": "Feature",
+                "properties": {},
+                "geometry": {
+                    "type": "MultiPolygon",
+                    "coordinates": [[[[8, 51], [11, 55], [12, 49], [8, 51]]]],
+                },
+            }
+        ],
+    }
+    # POST the new geostore
+    post_resp = await async_client.post("/geostore", json=payload)
+    assert post_resp.status_code == 201
+    assert post_resp.json()["data"]["gfw_geostore_id"] == expected_goestore_id
+    # Validate response structure
+    GeostoreResponse.parse_raw(post_resp.text)
+
+    # Do our GET again, which should now find the POSTed user area. Ensure
+    # the GET response is identical to the POST response
+    # (except for status code)
+    get_resp = await async_client.get(f"/geostore/{expected_goestore_id}")
+    assert get_resp.status_code == 200
+    assert get_resp.json() == post_resp.json()
+
+    # POSTing the same payload (more specifically geometry) again should yield
+    # the same response (we're squashing duplicate key errors)
+    post_resp2 = await async_client.post("/geostore", json=payload)
+    assert post_resp2.status_code == 201
+    assert post_resp2.json() == post_resp.json()
+
+
+@pytest.mark.asyncio
+async def test_dataset_version_geostore(async_client, batch_client):
+    _, logs = batch_client
+
+    ############################
+    # Setup test
+    ############################
+
+    dataset = "test"
+    source = GEOJSON_NAME
+    version = "v1.1.1"
+    input_data = {
+        "creation_options": {
+            "source_type": "vector",
+            "source_uri": [f"s3://{BUCKET}/{source}"],
+            "source_driver": "GeoJSON",
+            "zipped": False,
```

Comment:
I believe I removed the `zipped` attribute in my latest PR which is already in staging.

Reply:
Removed.